api/src/main/java/io/druid/segment/loading/DataSegmentArchiver.java

```diff
@@ -23,6 +23,25 @@
 
 public interface DataSegmentArchiver
 {
-  public DataSegment archive(DataSegment segment) throws SegmentLoadingException;
-  public DataSegment restore(DataSegment segment) throws SegmentLoadingException;
+  /**
+   * Perform an archive task on the segment and return the resulting segment or null if there was no action needed.
+   *
+   * @param segment The source segment
+   *
+   * @return The segment after archiving or `null` if there was no archiving performed.
+   *
+   * @throws SegmentLoadingException on error
+   */
+  DataSegment archive(DataSegment segment) throws SegmentLoadingException;
```

Comment:
Maybe annotate with `@javax.annotation.Nullable`? This will make IDE to warn you about using the object, returned from this method, without null check.